src/ssl.c

```diff
@@ -2884,6 +2884,14 @@ int wolfSSL_shutdown(WOLFSSL* ssl)
             }
         }
 
+#ifdef WOLFSSL_SHUTDOWNONCE
+        if (ssl->options.isClosed || ssl->options.connReset) {
+            /* Shutdown has already occurred.
+             * Caller is free to ignore this error. */
+            return PREV_SHUTDOWN;
```

Comment:
I am concerned "PREV_SHUTDOWN " is too generic and could collide with user apps. Can we use a return code like "SSL_SHUTDOWN_ALREADY_DONE_E"? Also this error code needs added to the internal.c `wolfSSL_ERR_reason_error_string` function.

Reply:
Good feedback. Thanks!